shiny/ui/_theme.py

```diff
@@ -457,9 +458,9 @@ def _html_dependency(self) -> HTMLDependency:
         css_name = self._dep_css_name()
 
         if self._css_temp_srcdir is None:
-            self._css_temp_srcdir = tempfile.TemporaryDirectory(delete=False)
+            self._css_temp_srcdir = tempfile.mkdtemp()
```

Comment:
How about a utility function that does `TemporaryDirectory(delete=False)` with python >= 3.12, and `mkdtemp()` otherwise? Feels like we'll run into this again, and it would be great to remember the right thing to do here.

Reply:
I like the idea, in particular because at some point when we're supporting only Python >= 3.12 we'd already have the right solution in place, but `tempfile.TemporaryDirectory()` returns a very different type of object (a `TemporaryDirectory()` instance) than `tempfile.mkdtemp()` (a `string`). Personally, I'm not sure that's worth the effort.

Reply:
Let's do `TemporaryDirectory()`, but keep a reference so that it doesn't get gc'd (when Python <3.12)